Automattic/jetpack
# tools: Refactor `--with` logic into composer-update-monorepo.sh

#### Changes proposed in this Pull Request:

In #25989 we added logic to version-packages.sh to use `--with` when calling composer-update-monorepo.sh to force monorepo deps in the "^1.2.3" style to install version 1.2.3 exactly, rather than picking up any later version, to better work in release branches where that style of dep is used.

And in #26072, we did the same thing for check-intra-monorepo-deps.sh for the same reason.

Chances are any other caller of composer-update-monorepo.sh is going to want the same too, so let's just move the logic there.

#### Other information:


#### Jetpack product discussion


p1662751617679859/1662483262.626649-slack-C01U2KGS2PQ

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Run version-packages.sh, as if to create a Jetpack release. See that things get updated as expected.
* Edit `projects/packages/videopress/CHANGELOG.md` to reflect a newer version than has actually been released. Run version-packages.sh, it should fail.
* Edit `projects/packages/videopress/CHANGELOG.md` to remove the newest changelog entry. Run version-packages.sh, it should succeed and should lock the videopress package to the older version in composer.lock.
* Check out c405506d5b6d1e4951d99aa188b918b4bb0f1448, cherry-pick the relevant changes (`git cherry-pick 7859236 ba6d621 f84d97c d2ed119`), then run `COMPOSER_ROOT_VERSION=dev-trunk tools/changelogger-release.sh packages/publicize -p`.
  * [ ] The dep on my-jetpack in `projects/plugins/jetpack/composer.json` should be changed to "^2.0.3".
  * [ ] Many other deps should change from "^1.2" to "^1.2.3" style.
  * [ ] The change in `projects/plugins/jetpack/composer.lock` is only to the record for the publicize package. No other packages change.
